addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/FuzzAPI.java

```diff
@@ -0,0 +1,842 @@
+        return getPayloadGeneratorMessageLocationList(location, start, end, payloadGenerator);
+    }
+
+    private List<PayloadGeneratorMessageLocation<?>> getPayloadGeneratorMessageLocationList(
+            HttpMessageLocation.Location location,
+            int start,
+            int end,
+            StringPayloadGenerator payloadGenerator) {
+        TextHttpMessageLocation messageLocation =
+                createTextHttpMessageLocationObject(start, end, location);
+        List<PayloadGeneratorMessageLocation<?>> payloadGeneratorMessageLocationList =
+                new ArrayList<>();
+        ResettableAutoCloseableIterator<DefaultPayload> resettableAutoCloseableIterator =
+                payloadGenerator.iterator();
+        PayloadGeneratorMessageLocation<?> payloadGeneratorMessageLocation =
+                new PayloadGeneratorMessageLocation<>(
+                        messageLocation,
+                        payloadGenerator.getNumberOfPayloads(),
+                        resettableAutoCloseableIterator);
+        payloadGeneratorMessageLocationList.add(payloadGeneratorMessageLocation);
+        return payloadGeneratorMessageLocationList;
+    }
+
+    private TextHttpMessageLocation createTextHttpMessageLocationObject(
+            int start, int end, HttpMessageLocation.Location location) {
+        return new TextHttpMessageLocation() {
+            @Override
+            public int getStart() {
+                return start;
+            }
+
+            @Override
+            public int getEnd() {
+                return end;
+            }
+
+            @Override
+            public Location getLocation() {
+                return location;
+            }
+
+            @Override
+            public Class<? extends Message> getTargetMessageClass() {
+                return HttpMessage.class;
+            }
+            // There is no need for these but can be fixed
+            // All of the functions below no need probably
+            @Override
+            public String getDescription() {
+                return (start + FUZZ_LOCATION_SEPARATOR + end);
+            }
+
+            @Override
+            public String getValue() {
+                return (start + FUZZ_LOCATION_SEPARATOR + end);
+            }
+
+            @Override
+            public boolean overlaps(MessageLocation otherLocation) {
+                return (((start + FUZZ_LOCATION_SEPARATOR + end)).equals(otherLocation.getValue()));
+            }
+
+            @Override
+            public int compareTo(MessageLocation messageLocation) {
+                return (messageLocation
+                        .getValue()
+                        .compareTo(((start + FUZZ_LOCATION_SEPARATOR + end))));
+            }
+        };
+    }
+
+    /**
+     * Method copied from HttpFuzzerHandler
+     *
+     * @see HttpFuzzerHandler /* Used to create the fuzzer
+     */
+    /* Didn't want to change fuzzerHandler to public so reusing it */
```

Comment:
These should be combined and formatted properly.

Is there a reason it shouldn't be public? (Or protected)

Reply:
Yeah there is no reason for it to be not public. 
Just wanted to confirm.

Reply:
@thc202 / @psiinon thoughts on making it public?

Reply:
It's fine making it public, noting that's not part of the API.